### DISCUSSION TITLE
Constant requests?

### DISCUSSION THREAD

**Discussion by adamesq** [gildas-lormeau/SingleFile#1229] [Q&A]
Guessing I have something set wrong here. 

SingleFile is constantly requesting access to every new page I run across. I feel that the extension should ONLY need access when I'm clicking SingleFile to DO something...not when it's just sitting there unused in my Safari bar. Is this as settings issue?

(ie. The extension "SingleFile" would like to access zzzzz.com, yyyy.com, and 3 other websites. This extension would be able to read and alter webpages and see your browsing history on these websites. This could include sensitive information, including passwords, phone numbers, and credit cards.)

**Comment by gildas-lormeau** [comment]
That's indeed unexpected. What OS are you using, i.e. macOS or iOS? Did you try to reset the settings in the options page of SingleFile?

**Reply by adamesq** [reply]
Thank you for the quick reply, @gildas-lormeau! MacS Ventura.  I've just reset after reading!🤞🏻

**Reply by gildas-lormeau** [reply]
Did resetting the options fix the problem?

**Reply by adamesq** [reply]
Resetting didn't fix the issue. I'm seeing Singlefile alerts randomly pop up for sites. For example, I just went to ideogram.ai and got the attached popup. My mouse was nowhere near the Silglefile extension. Meanwhile, the extension has the warning symbol and indicates "The extension "SingleFile" would like to access additional websites." WHY? I'm not asking it to do anything. 🤔

**Reply by gildas-lormeau** [reply]
That's because SingleFile needs to inject a script in pages before any other script mainly in order to detect fonts loaded with the [`FontFace`](https://developer.mozilla.org/docs/Web/API/FontFace) API. Unfortunately, there is no other technical solution to fix this. I'll do some tests in Safari.

**Reply by adamesq** [reply]
Hmmm...that makes sense, but can't this only happen when the extension is clicked (maybe trigger a reload)?

**Reply by gildas-lormeau** [reply]
The script must be injected before any other scripts on the page. Otherwise it won't work.

**Reply by adamesq** [reply]
The script loads on every single site visited even if the extension isn't being clicked or used at all?

**Reply by gildas-lormeau** [reply]
Yes, otherwise it wouldn't work.
